File: numeric/euclidgcd/euclidgcd.m
#import <Foundation/Foundation.h>
#include <stdlib.h>

int euclidgcd(int m, int n) {
    int r = 0;
    while (n != 0) {
        r = m % n;
        m = n;
        n = r;
    }
    return m;
}

int main(int argc, const char * argv[]) {
    int m = 15;
    int n = 10;

    if (argc >= 2) {
        m = atoi(argv[1]);
        n = atoi(argv[2]);
    }

    @autoreleasepool {
        NSLog(@"%d %d", m, n);
        NSLog(@"gcd: %d", euclidgcd(m, n));
    }
    return 0;
}

File: numeric/euclidgcd/euclidgcd.cpp
#include <print>
#include <cstdlib>

int euclidgcd(int m, int n)
{
    int r = m % n;
    while (r != 0)
    {
        m = n;
        n = r;
        r = m % n;
    }
    return n;
}

int main(int argc, char *argv[])
{
    int v_1 = 10;
    int v_2 = 15;

    if (argc >= 2)
    {
        v_1 = std::stoi(argv[1]);
        v_2 = std::stoi(argv[2]);
    }

    std::println("{} {}", v_1, v_2);
    std::println("{}", euclidgcd(v_1, v_2));

    return 0;
}
